src/main/scala/org/apache/mesos/chronos/scheduler/jobs/TaskManager.scala

```diff
@@ -41,8 +42,7 @@ class TaskManager @Inject()(val listeningExecutor: ListeningScheduledExecutorSer
 
   val taskCache = CacheBuilder.newBuilder().maximumSize(5000L).build[String, TaskState]()
 
-  val taskMapping = new mutable.HashMap[String, mutable.ListBuffer[(String, Future[_])]] with
-    collection.mutable.SynchronizedMap[String, mutable.ListBuffer[(String, Future[_])]]
+  val taskMapping: concurrent.Map[String, mutable.ListBuffer[(String, Future[_])]] = new ConcurrentHashMap().asScala
```

Comment:
I agree that this should be updated to be a concurrent hash map, but this probably means we can get rid of some of the synchronized blocks in this class. Perhaps this should be a separate PR.